cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/BuiltInRolesAdministrationTest.scala

```diff
@@ -0,0 +1,329 @@
+
+  override def doc: Document = new DocBuilder {
+    doc("Built-in roles", "administration-built-in-roles")
+    database("system")
+    synopsis("This section explains the default privileges of the built-in roles in Neo4j and how to recreate them if needed.")
+
+    p(
+      """All of the commands described in the enclosing section require that the user executing the commands has the rights to do so.
+        |The privileges listed in the following sections are the default set of privileges for each built-in role.
+        |""".stripMargin)
+    p(
+      """
+        |* <<administration-built-in-roles-public, The `PUBLIC` role>>
+        |* <<administration-built-in-roles-reader, The `reader` role>>
+        |* <<administration-built-in-roles-editor, The `editor` role>>
+        |* <<administration-built-in-roles-publisher, The `publisher` role>>
+        |* <<administration-built-in-roles-architect, The `architect` role>>
+        |* <<administration-built-in-roles-admin, The `admin` role>>
+        |""".stripMargin)
+    section("The `PUBLIC` role", "administration-built-in-roles-public", "enterprise-edition") {
+      p("All users are granted the `PUBLIC` role. It cannot be revoked or dropped. By default it gives access to the default database.")
+      section("Privileges of the `PUBLIC` role", "administration-built-in-roles-public-privileges", "enterprise-edition") {
+        query("SHOW ROLE PUBLIC PRIVILEGES", ResultAssertions(p => true)) {
+          resultTable()
+        }
+      }
+      section("How to recreate the `PUBLIC` role", "administration-built-in-roles-public-recreate", "enterprise-edition") {
+        initQueries(
+          """REVOKE GRANT ACCESS ON DEFAULT DATABASE FROM PUBLIC""")  // setup so that later when the grant query gets executed, it will show systemUpdates: 1
+        p(
+          """The `PUBLIC` role can not be dropped and thus there is no need to recreate the role itself.
+            |To restore the role to its original capabilities two steps are needed: First, all `GRANT` or `DENY` on this role should be revoked (see output of `SHOW ROLE PUBLIC PRIVILEGES` on what to revoke).
+            |Secondly, the following query need to be executed:""".stripMargin)
+        query("GRANT ACCESS ON DEFAULT DATABASE TO PUBLIC", ResultAssertions((r) => {
+          assertStats(r, systemUpdates = 1)
+        })) {
+          statsOnlyResultTable()
+        }
+
+        p("The resulting `PUBLIC` role has the same privileges as the " +
+          "<<administration-built-in-roles-public-privileges,original built-in `PUBLIC` role>>.")
+
+//        query("SHOW ROLE PUBLIC PRIVILEGES", ResultAssertions(p => true)) {
+//          resultTable()
+//        }
+      }
+    }
+    section("The `reader` role", "administration-built-in-roles-reader", "enterprise-edition") {
+      p("The `reader` can perform read-only queries on all databases except `system`.")
```

Comment:
This is a bit tricky. Since we are showing the match privileges below, which is on `GRAPH *`, we should maybe be talking about graphs rather than databases here. On the other hand I think it is useful information that we cannot read on system and we don't really have the concept of a system graph externally, only system database